src/lightning/fabric/loggers/csv_logs.py

```diff
@@ -235,6 +235,7 @@ def save(self) -> None:
         new_keys = self._record_new_keys()
         file_exists = self._fs.isfile(self.metrics_file_path)
 
+        self.metrics_keys = sorted(self.metrics_keys)
         if new_keys and file_exists:
             # we need to re-write the file if the keys (header) change
             self._rewrite_with_new_header(self.metrics_keys)
```

Comment:
Sure, we can. But I could see L241 which conditionally writes a log file via `_rewrite_with_new_header` inside which again CSV files are written where the header is used from `self.metric_keys`.  So, I thought to sort the column headers globally so that it will be consistent for all the csv logs wherever it's used.
![image](https://github.com/Lightning-AI/pytorch-lightning/assets/43300971/af28405b-ffb2-4686-badc-587882d02da2)

Let me know if it's not the case. I will modify the code as per your recommendation. :)

Reply:
In this case, my new suggestion would be to do the sorting in the `_record_new_keys` method.

Reply:
Cool. Refactored.